packages/tendermint-rpc/src/rpcclients/httpbatchclient.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  isJsonRpcErrorResponse,
+  JsonRpcRequest,
+  JsonRpcSuccessResponse,
+  parseJsonRpcResponse,
+} from "@cosmjs/json-rpc";
+
+import { http, HttpEndpoint } from "./httpclient";
+import { hasProtocol, RpcClient } from "./rpcclient";
+
+export interface HttpClientOptions {
+  dispatchInterval: number;
+  batchSizeLimit: number;
+}
+
+export const defaultHttpClientOptions: HttpClientOptions = { dispatchInterval: 20, batchSizeLimit: 20 };
+
+export class HttpBatchClient implements RpcClient {
+  protected readonly url: string;
+  protected readonly headers: Record<string, string> | undefined;
+  protected readonly options: HttpClientOptions;
+
+  private readonly queue: Array<{
+    request: JsonRpcRequest;
+    resolve: (a: JsonRpcSuccessResponse) => void;
+    reject: (a: Error) => void;
+  }> = [];
+
+  public constructor(endpoint: string | HttpEndpoint, options: HttpClientOptions = defaultHttpClientOptions) {
+    this.options = options;
+    if (typeof endpoint === "string") {
+      // accept host.name:port and assume http protocol
+      this.url = hasProtocol(endpoint) ? endpoint : "http://" + endpoint;
+    } else {
+      this.url = endpoint.url;
+      this.headers = endpoint.headers;
+    }
+    setInterval(() => this.tick(), options.dispatchInterval);
+  }
+
+  public disconnect(): void {
+    // nothing to be done
+  }
+
+  public async execute(request: JsonRpcRequest): Promise<JsonRpcSuccessResponse> {
+    return new Promise((resolve, reject) => {
+      this.queue.push({ request, resolve, reject });
+    });
+  }
+
+  private async tick(): Promise<void> {
+    // Avoid race conditions
+    const queue = this.queue.splice(0, this.options.batchSizeLimit);
```

Comment:
Can you add a check that `batchSizeLimit` is set and >= 1? Just throw if that's not the case.

I would like to avoid running into special behaviour for the decond argument of `splice`:

>  If deleteCount is 0 or negative, no elements are removed. In this case, you should specify at least one new element (see below).